test/e2e/operator_test.go

```diff
@@ -455,118 +280,76 @@ u3YLAbyW/lHhOCiZu2iAI8AbmXem9lW6Tr7p/97s0w==
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
 	}
 
 	// Scale back down.
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller: %v", err)
 	}
-	ic.Spec.Replicas = &originalReplicas
-	if err := cl.Update(context.TODO(), ic); err != nil {
+	ic.Spec.Replicas = originalReplicas
+	if err := kclient.Update(context.TODO(), ic); err != nil {
 		t.Fatalf("failed to update ingresscontroller: %v", err)
 	}
 
 	// Wait for the deployment scale down to be observed.
-	err = waitForAvailableReplicas(cl, name, 2*time.Minute, originalReplicas)
-	if err != nil {
-		t.Fatalf("failed waiting deployment %s to scale to %d: %v", deploymentName, originalReplicas, err)
+	if err := waitForAvailableReplicas(kclient, ic, 2*time.Minute, *originalReplicas); err != nil {
+		t.Fatalf("failed waiting deployment %s to scale to %d: %v", defaultName, originalReplicas, err)
 	}
 
 	// Ensure the ingresscontroller remains available
-	err = waitForIngressControllerCondition(cl, 2*time.Minute, name, defaultAvailableConditions...)
-	if err != nil {
-		t.Errorf("failed to observe expected conditions: %v", err)
-	}
-}
-
-func waitForAvailableReplicas(cl client.Client, name types.NamespacedName, timeout time.Duration, expectedReplicas int32) error {
-	ic := &operatorv1.IngressController{}
-	var lastObservedReplicas int32
-	err := wait.PollImmediate(1*time.Second, timeout, func() (bool, error) {
-		if err := cl.Get(context.TODO(), name, ic); err != nil {
-			return false, nil
-		}
-		lastObservedReplicas = ic.Status.AvailableReplicas
-		if lastObservedReplicas != expectedReplicas {
-			return false, nil
-		}
-		return true, nil
-	})
-	if err != nil {
-		return fmt.Errorf("failed to achieve expected replicas: last observed : %v", lastObservedReplicas)
+	if err := waitForIngressControllerCondition(kclient, 5*time.Minute, defaultName, defaultAvailableConditions...); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
 	}
-	return nil
 }
 
 func TestRouterCACertificate(t *testing.T) {
-	cl, ns, err := getClient()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	name := types.NamespacedName{Namespace: ns, Name: manifests.DefaultIngressControllerName}
 	ic := &operatorv1.IngressController{}
-	if err := cl.Get(context.TODO(), name, ic); err != nil {
-		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	if err := kclient.Get(context.TODO(), defaultName, ic); err != nil {
+		t.Fatalf("failed to get default ingresscontroller: %v", err)
 	}
 
 	if ic.Status.EndpointPublishingStrategy.Type != operatorv1.LoadBalancerServiceStrategyType {
-		t.Skip("test skipped for non-cloud HA type")
+		t.Skip("test only applicable to load balancer controllers")
```

Comment:
This skip makes me realize that all tests without a skip that expect `defaultAvailableConditions` should be failing on non-cloud platforms.